presto-main/src/test/java/com/facebook/presto/memory/TestMemoryPools.java

```diff
@@ -226,6 +230,50 @@ public void testBlockingOnRevocableMemoryFreeViaRevoke()
         assertEquals(userPool.getFreeBytes(), 2);
     }
 
+    @Test
+    public void testTaggedAllocations()
+    {
+        QueryId testQuery = new QueryId("test_query");
+        MemoryPool testPool = new MemoryPool(new MemoryPoolId("test"), new DataSize(1000, BYTE));
+
+        testPool.reserve(testQuery, Optional.of("test_tag"), 10);
+
+        Map<String, Long> allocations = testPool.getTaggedMemoryAllocations().get(testQuery);
+        assertTaggedAllocations(allocations, new String[] {"test_tag"}, new long[] {10});
```

Comment:
This seems equivalent to `assertEquals(allocations, ImmutableMap.of("test_tag", 10L))`